integration_tests/test_model_upload.py

```diff
@@ -28,12 +30,26 @@ def setUp(self) -> None:
     def test_model_upload_and_versioning(self) -> None:
         # Create Instance
         model_upload(self.handle, self.temp_dir, LICENSE_NAME)
+        time.sleep(5)  # It takes time for the instance to be created
```

Comment:
I'm guessing you didn't mean to keep this?

Reply:
no i did, its because it takes a little bit of time for the api to work on a newly uploaded model, meaning if you upload a model X, and call api.get(x) right away, it will say x not found

Reply:
The backend waits for the instance's archive to be created. But waiting 5 seconds won't work all the time.
What about trying to get the model until it works instead?

Reply:
the get here is called inside upload_model(), and waiting does not work because sometimes the model actually does not exist. how about I wait for a longer period of time? e.g. 60 seconds

Reply:
Waiting for a number of seconds in any code is too flaky to be a good solution unfortunately.

Reply:
> @Philmod - do we keep a status to indicate what state the model is in its lifecycle?

Reply:
infinite loop is also not good because if the model actually doesnt exist, it would loop infinitely, would looping for a maximum number of retries with a delay between each try be better for the test? if not, do we know when the instance's archive updates or is it a random event?

Reply:
```
def create_model_instance_or_version(
    model_handle: ModelHandle, files: List[str], license_name: Optional[str], version_notes: str = ""
) -> None:
    try:
        api_client = KaggleApiV1Client()
        api_client.get(f"/models/{model_handle}/get", model_handle)
        # the instance exist, create a new version.
        _create_model_instance_version(model_handle, files, version_notes)
    except KaggleApiHTTPError as e:
        if e.response is not None and (
            e.response.status_code == HTTPStatus.NOT_FOUND  # noqa: PLR1714
            or e.response.status_code == HTTPStatus.FORBIDDEN
        ):
            _create_model_instance(model_handle, files, license_name)
        else:
            raise (e)
```
so `model_upload` creates an instance or a variation depending on wether an instance already exists for a model. having a loop around the `api_client.get` here would break that logic

Reply:
a reasonable approach could be to loop in the test case around the second `model_upload` call until it finds the model with:
```
max_retries = 5 
retry_delay = 10 
```